File: codility/src/main/java/codility/com/lessons/l04countingelements/MissingInteger.java
package codility.com.lessons.l04countingelements;

import java.util.Collection;
import java.util.HashSet;

/**
 * 100%
 * @author 08115341622
 */
public class MissingInteger {
	public int solution(int[] A) {
		Collection<Integer> numeros = new HashSet<Integer>();
		for (int x : A) {
			numeros.add(x);
		}
		int min = 1;
		while (numeros.contains(min)) {
			min++;
		}
		return min;
	}
}

File: codility_new/missing_integer.py
# https://app.codility.com/programmers/lessons/4-counting_elements/missing_integer/
def solution(A):
    A = sorted(filter(lambda x: x > 0, set(A)))
    if len(A) == 0 or A[0] != 1:
        return 1
    
    for i in range(1, len(A)):
        if A[i] != A[i-1] + 1:
            return A[i-1] + 1
    
    return A[-1] + 1
